cache_utils/decorators.py

```diff
@@ -2,17 +2,51 @@
 
 import logging
 
-from cache_utils.utils import _cache_key, _func_info, _func_type, sanitize_memcached_key
-
 from django.core.cache import caches
+from django.utils.encoding import smart_text
 from django.utils.functional import wraps
 
+from cache_utils.utils import (
+    serialize,
+    sanitize_memcached_key,
+    PREFIX,
+)
+import six
+
 
 logger = logging.getLogger("cache_utils")
 
 
-def cached(timeout, group=None, backend=None, key=None):
-    """ Caching decorator. Can be applied to function, method or classmethod.
+def default_key(*args, **kwargs):
+    return args, kwargs
+
+
+def default_fn_key(func):
+    return ".".join([func.__module__, func.__qualname__])
```

Comment:
I like this, didn't like lineno either, and qualname is better. But I suppose we'll have to make sure to keep on using the `legacy` ones for the caches we've already set up.